graphql-server/src/schemas/custom.ts

```diff
@@ -68,15 +135,31 @@ export async function buildCustomSchema(ogm: OGM) {
           sort,
           limit,
           offset,
+          phiEnabled,
         }: QueryDashboardPatientsArgs
       ) {
         const queryBody = buildPatientsQueryBody({ searchVals, columnFilters });
-        return await queryDashboardPatients({
+        const queryFinal = buildPatientsQueryFinal({
           queryBody,
           sort,
           limit,
           offset,
         });
+        const patientsData = await queryDashboardPatients(queryFinal);
+        if (!phiEnabled || !searchVals || searchVals?.length == 0) {
+          return patientsData;
+        }
+        const patientIdsTriplets = await queryPatientIdsTriplets(searchVals);
+        const dmpPatientIds = patientIdsTriplets
+          .map((t) => t.DMP_PATIENT_ID)
+          .filter((id): id is string => !!id);
+        const anchorSeqDatesByDmpPatientId =
+          await queryAnchorSeqDatesByDmpPatientId(dmpPatientIds);
+        return mapPhiToPatientsData({
+          patientsData,
+          patientIdsTriplets,
+          anchorSeqDatesByDmpPatientId,
+        });
```

Comment:
If PHI is disabled or no search values are provided, run and return the patients data from Neo4j as before.

Otherwise, query the ID mapping from Databricks > use the DMP Patient IDs returned to query the corresponding anchor sequencing dates > map PHI to the Neo4j patients data.

Reply:
TODO: Right now, `anchorSequencingDate` will be returned only if the corresponding DMP Patient IDs are found in both the ID mapping table *and* Neo4j. Instead of using DMP Patient IDs from the ID mapping results to query sequencing dates, use DMP Patient IDs from Neo4j instead.